File: @@Cohort[10]-Code/NicholasWertman-java/module-1/10_Classes_Encapsulation/student-exercise/java/src/main/java/com/techelevator/HomeworkAssignment.java
package com.techelevator;

public class HomeworkAssignment {

	private int totalMarks;
	private int possibleMarks;
	private String submitterName;
	private String letterGrade;

	public HomeworkAssignment(int possibleMarks) {
		this.possibleMarks = possibleMarks;
		this.totalMarks = totalMarks;
		this.submitterName = submitterName;
		this.letterGrade = letterGrade;
				
	}

	/**
	 * @return the totalMarks
	 */
	public int getTotalMarks() {
		return totalMarks;
	}

	/**
	 * @param totalMarks the totalMarks to set
	 */
	public void setTotalMarks(int totalMarks) {
		this.totalMarks = totalMarks;
	}

	/**
	 * @return the submitterName
	 */
	public String getSubmitterName() {
		return submitterName;
	}

	/**
	 * @param submitterName the submitterName to set
	 */
	public void setSubmitterName(String submitterName) {
		this.submitterName = submitterName;
	}

	/**
	 * @return the possibleMarks
	 */
	public int getPossibleMarks() {
		return possibleMarks;
	}

	/**
	 * @return the letterGrade
	 */
	//public String getLetterGrade() {
	//	return letterGrade;
	//}

	public String getLetterGrade() {
		
		double percentage =  (possibleMarks/ totalMarks)* 100;
		
		if ((percentage >= 90)) {
			return "A";
		} else 
			if (( percentage >= 80 && ( percentage < 90))) {
			 return "B";
		} else 
			if ((percentage >= 70 && (percentage < 80))) {
			return "C";
		} else
			if ((percentage >= 60 && (percentage < 70))) {
			return "D";
		} else {
			return "F";
		}
	}
}

File: java-mvc-4-mariah/module-1/10_Classes_Encapsulation/exercise-final/dotnet/Individual.Exercises/Classes/HomeworkAssignment.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Individual.Exercises.Classes
{
    public class HomeworkAssignment
    {
        

        /// <summary>
        /// Total number of marks received
        /// </summary>
        public int TotalMarks { get; set; }
                

        /// <summary>
        /// Possible number of marks to get right on the homework assignment.
        /// </summary>
        public int PossibleMarks { get; }
        
        

        /// <summary>
        /// Name of the person who submitted the homework assignment
        /// </summary>
        public string SubmitterName { get; }
        

        /// <summary>
        /// Letter grade for the assignment. (90-100 A, 80-89 B, 70-79 C, 60-69 D, otherwise F)
        /// </summary>
        public string LetterGrade
        {
            get
            {                
                double percentage = (double)TotalMarks / PossibleMarks;
                if (percentage >= 0.9)
                {
                    return "A";
                }
                else if (percentage >= 0.8)
                {
                    return "B";
                }
                else if (percentage >= 0.7)
                {
                    return "C";
                }
                else if (percentage >= 0.6)
                {
                    return "D";
                }
                else
                {
                    return "F";
                }
            }
            
        }

        /// <summary>
        /// Homework assignment requires possible marks
        /// </summary>
        /// <param name="possibleMarks"></param>
        public HomeworkAssignment(int possibleMarks, string submitterName)
        {
            this.PossibleMarks = possibleMarks;
            this.SubmitterName = submitterName;
        }


    }
}
